File: 1252-cells-with-odd-values-in-a-matrix/1252-cells-with-odd-values-in-a-matrix.cpp
class Solution {
public:
    int oddCells(int m, int n, vector<vector<int>>& indices) {
        vector<int> row_inc(m, 0);
        vector<int> col_inc(n, 0);
        
        for (auto& idx : indices) {
            row_inc[idx[0]]++;
            col_inc[idx[1]]++;
        }
        
        int odd_count = 0;
        for (int i = 0; i < m; ++i) {
            for (int j = 0; j < n; ++j) {
                if ((row_inc[i] + col_inc[j]) % 2 == 1) {
                    odd_count++;
                }
            }
        }
        return odd_count;
    }
};

File: 1252-cells-with-odd-values-in-a-matrix/1252-cells-with-odd-values-in-a-matrix.java
class Solution {
    public int oddCells(int m, int n, int[][] indices) {
        int[] row = new int[m];
        int[] col = new int[n];

        // Count increments for each row and column
        for (int[] index : indices) {
            row[index[0]]++;
            col[index[1]]++;
        }

        // Count how many rows and columns have odd increments
        int oddRows = 0;
        for (int r : row) {
            if (r % 2 == 1) oddRows++;
        }

        int oddCols = 0;
        for (int c : col) {
            if (c % 2 == 1) oddCols++;
        }

        // Cells that are odd = (oddRows * evenCols) + (evenRows * oddCols)
        return oddRows * (n - oddCols) + (m - oddRows) * oddCols;
    }
}